src/ol/renderer/webgl/VectorLayer.js

```diff
@@ -0,0 +1,344 @@
+ * @property {Array<CustomAttribute>} [attributes] These attributes will be read from the features in the source
+ * and then passed to the GPU. The `name` property of each attribute will serve as its identifier:
+ *  * In the vertex shader as an `attribute` by prefixing it with `a_`
+ *  * In the fragment shader as a `varying` by prefixing it with `v_`
+ * Please note that these can only be numerical values.
+ * @property {string} polygonVertexShader Vertex shader source, mandatory.
+ * @property {string} polygonFragmentShader Fragment shader source, mandatory.
+ * @property {string} lineStringVertexShader Vertex shader source, mandatory.
+ * @property {string} lineStringFragmentShader Fragment shader source, mandatory.
+ * @property {string} pointVertexShader Vertex shader source, mandatory.
+ * @property {string} pointFragmentShader Fragment shader source, mandatory.
+ * @property {string} [hitVertexShader] Vertex shader source for hit detection rendering.
+ * @property {string} [hitFragmentShader] Fragment shader source for hit detection rendering.
+ * @property {Object<string,import("../../webgl/Helper").UniformValue>} [uniforms] Uniform definitions for the post process steps
+ * Please note that `u_texture` is reserved for the main texture slot.
+ * @property {Array<import("./Layer").PostProcessesOptions>} [postProcesses] Post-processes definitions
+ */
+
+/**
+ * @classdesc
+ * Experimental WebGL vector renderer. Supports polygons and lines.
+ *
+ * You need to provide vertex and fragment shaders for rendering. This can be done using
+ * {@link module:ol/webgl/ShaderBuilder} utilities.
+ *
+ * To include variable attributes in the shaders, you need to declare them using the `attributes` property of
+ * the options object like so:
+ * ```js
+ * new WebGLPointsLayerRenderer(layer, {
+ *   attributes: [
+ *     {
+ *       name: 'size',
+ *       callback: function(feature) {
+ *         // compute something with the feature
+ *       }
+ *     },
+ *     {
+ *       name: 'weight',
+ *       callback: function(feature) {
+ *         // compute something with the feature
+ *       }
+ *     },
+ *   ],
+ *   vertexShader:
+ *     // shader using attribute a_weight and a_size
+ *   fragmentShader:
+ *     // shader using varying v_weight and v_size
+ * ```
+ *
+ * To enable hit detection, you must as well provide dedicated shaders using the `hitVertexShader`
+ * and `hitFragmentShader` properties. These shall expect the `a_hitColor` attribute to contain
+ * the final color that will have to be output for hit detection to work.
+ *
+ * The following uniform is used for the main texture: `u_texture`.
+ *
+ * Please note that the main shader output should have premultiplied alpha, otherwise visual anomalies may occur.
+ *
+ * Polygons are broken down into triangles using the @mapbox/earcut package.
+ * Lines are rendered into strips of quads.
+ *
+ *
+ * This uses {@link module:ol/webgl/Helper~WebGLHelper} internally.
+ *
+ * @api
```

Comment:
I think we should keep this "experimental" (not tagged with `@api`) for now.

Reply:
Totally agree, I'll change it; guess I wasn't really paying attention to API consistency while working on this, as it is obviously in the experimental field for now

Reply:
I pushed an additional commit to remove this `@api` annotation.